packages/wallet/src/errors.ts

```diff
@@ -0,0 +1,55 @@
+import { inspect } from "node:util";
+
+export class InvalidPayloadError extends Error {
+    constructor(public payload: unknown) {
+        super(`Invalid payload: ${inspect(payload)}`);
+    }
+}
+
+export class RelayError extends Error {
+    constructor() {
+        super(
+            `You need to call the method relayDAppAddress from DAppAddressRelay__factory.`,
+        );
+    }
+}
+
+export class InsufficientBalanceError extends Error {
+    constructor(user: string, token: string, amount: bigint, tokenId?: bigint) {
+        super(
+            `insufficient balance of user ${user} of token ${token}: ${amount.toString()}${
+                tokenId ? ` of tokenId ${tokenId}` : ""
+            }`,
+        );
+    }
+}
+
+export class WalletUndefinedError extends Error {
+    constructor(address: string) {
+        super(`wallet of user ${address} is undefined`);
+    }
+}
+
+export class TokenFromUserNotFound extends Error {
+    constructor(user: string, token: string, tokenId: bigint) {
+        super(`user ${user} does not have token ${tokenId} of token ${token}`);
+    }
+}
+
+export class ArrayNoSameLength extends Error {
+    constructor(first: string, second: string) {
+        super(`${first} and ${second} must have the same length`);
+    }
+}
+
+export class NegativeTokenIdError extends Error {
+    constructor(tokenId: bigint, amount: bigint) {
+        super(`negative value for tokenId ${tokenId}: ${amount}`);
```

Comment:
are we raising this in what situation? tokenId is never negative as it comes from a uint256

Reply:
Renaming for when the values passed are negative in the `amount` and not in the `tokenId`.